cg/meta/store/balsamic.py

```diff
@@ -57,36 +59,55 @@ def _create_analysis(bundle_data, case_obj, status, version_obj):
 
 
 def _reset_analysis_action(case_obj):
+    """Resets desired action on the case"""
     case_obj.action = None
 
 
-def _add_analysis(config_stream, case_obj):
+def parse_created(config_data: dict) -> dt.datetime:
+    """Parse out analysis creation data from analysis"""
+    datetime_str = config_data["analysis"]["config_creation_date"]
+    return dt.datetime.strptime(datetime_str, "%Y-%m-%d %H:%M")
+
+
+def parse_version(config_data: dict) -> dt.datetime:
+    """Parse out analysis pipeline version from analysis"""
+    return config_data["analysis"]["BALSAMIC_version"]
+
+
+def _parse_bundle_data(config_path, deliverables_file, case_obj):
     """Gather information from balsamic analysis to store."""
-    with Path(config_stream).open() as in_stream:
-        meta_raw = ruamel.yaml.safe_load(in_stream)
+
+    with Path(config_path).open() as config_stream:
+        config_raw = ruamel.yaml.safe_load(config_stream)
+        created = parse_created(config_raw)
+        version = parse_version(config_raw)
+
+    with Path(deliverables_file).open() as in_stream:
+        deliverables_raw = ruamel.yaml.safe_load(in_stream)
+
     new_bundle = _build_bundle(
-        meta_raw, name=case_obj.internal_id, created=dt.datetime.now(), version="1"
+        deliverables_raw, name=case_obj.internal_id, created=created, version=version
     )
     return new_bundle
 
 
-def _build_bundle(meta_data: dict, name: str, created: dt.datetime, version: str) -> dict:
+def _build_bundle(deliverables_data: dict, name: str, created: dt.datetime, version: str) -> dict:
```

Comment:
These are also very general functions that many parts of the code could make use of